### PR TITLE
Feat/#15 set order mass

### PR DESCRIPTION
**Pull request by Poukam-Pierre**

### CODE REVIEW

File: apps/easy-messe/components/OfferMass/LetOfferMass.tsx [hunk 1]
```diff
@@ -0,0 +1,343 @@
...
+                        variant="outlined"
+                        sx={{
+                            minWidth: { laptop: 'initial', mobile: '145px' }
+                        }}
+                    >
+                        {formatMessage({ id: label })}
+                    </Button>
+                ))}
+            </Box>
+            <Typography variant="h2">{formatMessage({ id: 'ApplicantInformation' })}</Typography>
```
Comment (Tchakoumi) [inline apps/easy-messe/components/OfferMass/LetOfferMass.tsx:191]:
```suggestion
            <Typography variant="h2">{formatMessage({ id: 'applicantInformation' })}</Typography>
```

these data should be in camelCase

File: apps/easy-messe/components/OfferMass/LetOfferMass.tsx [hunk 2]
```diff
@@ -0,0 +1,343 @@
...
+                padding: {
+                    laptop: '10px 10px 20px 0',
+                    mobile: '10px 0 20px 0'
+                }
+            }}>
+                <FormControlLabel
+                    control={
+                        <Switch
+                            checked={isAnonym}
+                            onChange={(_) => setIsAnonym(_.target.checked)}
```
Comment (Tchakoumi) [inline apps/easy-messe/components/OfferMass/LetOfferMass.tsx:205]:
```suggestion
                            onChange={(event) => setIsAnonym(event.target.checked)}
```

File: apps/easy-messe/components/OfferMass/LetOfferMass.tsx [hunk 3]
```diff
@@ -0,0 +1,343 @@
...
+                            <TextField
+                                placeholder={formatMessage({ id: placeholder })}
+                                size="small"
+                                disabled={isAnonym}
+                            />
+                        </Box>
+                    ))}
+                </Box>
+            </Box>
+            <Typography variant="h2">{formatMessage({ id: 'MassInformations' })}</Typography>
```
Comment (Tchakoumi) [inline apps/easy-messe/components/OfferMass/LetOfferMass.tsx:237]:
```suggestion
            <Typography variant="h2">{formatMessage({ id: 'massInformations' })}</Typography>
```

File: apps/easy-messe/components/OfferMass/LetOfferMass.tsx
Comment (Tchakoumi) [inline apps/easy-messe/components/OfferMass/LetOfferMass.tsx:1]: verify all your formatMessage texts and make them in camelCase.

**Review by Tchakoumi — CHANGES_REQUESTED** [review]

File: apps/easy-messe/components/OfferMass/LetOfferMass.tsx [hunk 4]
```diff
@@ -0,0 +1,343 @@
...
+                },
+                {
+                    price: 3000,
+                    dateTime: new Date('2024-06-16T17:00:00'),
+                    massType: null
+                },
+            ]
+        },
+    ]
+    const massLoop: MassGroupCategory[] = [
```
[the remark this answers is not in the document]
Comment (Poukam-Pierre) [reply apps/easy-messe/components/OfferMass/LetOfferMass.tsx:122]: Here is different kind of mass group. In simple click, we can offer a group of masses througth the calendar on specific church.

[the remark this answers is not in the document]
Comment (Poukam-Pierre) [reply apps/easy-messe/components/OfferMass/LetOfferMass.tsx:122]: I've changed massLoop by massOrderCategory

File: apps/easy-messe/components/OfferMass/DateTimeMass/DateTimeMass.tsx [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+
+    const isTimeAllowed = (time: Date) => {
+        if (selectedDate) {
+            const dateSelected = allMassDates?.filter((date) => isSameDay(date, selectedDate.toDate()))
+            return dateSelected?.some((allowedTime) =>
+                allowedTime.getHours() === time.getHours() && allowedTime.getMinutes() === time.getMinutes())
+        }
+    }
+    return (
+        <LocalizationProvider dateAdapter={AdapterDayjs} adapterLocale={activeLanguage}>
```
[the remark this answers is not in the document]
Comment (Poukam-Pierre) [reply apps/easy-messe/components/OfferMass/DateTimeMass/DateTimeMass.tsx:55]: It's only appear in this part of app.  But we'll also use that on admin App. As a monoripo project is there good to still send this provider at the root?

File: apps/easy-messe/components/OfferMass/DateTimeMass/DateTimeMass.tsx [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+import { isSameDay } from 'date-fns/isSameDay';
+import Calendar from './Calendar';
+import Time from './TimeClock';
+
+
+export default function DateTimeMassPicker({
+    parishDataFetch
+}: {
+    parishDataFetch: ParishDataFetch | undefined
+}) {
```
[the remark this answers is not in the document]
Comment (Poukam-Pierre) [reply apps/easy-messe/components/OfferMass/DateTimeMass/DateTimeMass.tsx]: review0
There is a problem when put the type straight in. Can you show me the way?

[the remark this answers is not in the document]
Comment (Tchakoumi) [reply apps/easy-messe/components/OfferMass/DateTimeMass/DateTimeMass.tsx]: It should work.
verify what attributes exist in the ParishData and use it.

you cannot try to destructure `parishData` from `ParishData`, it does not make sense

[the remark this answers is not in the document]
Comment (Poukam-Pierre) [reply apps/easy-messe/components/OfferMass/DateTimeMass/DateTimeMass.tsx]: yes it's true. haha..

File: apps/easy-messe/components/OfferMass/DateTimeMass/DateTimeMass.tsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (Poukam-Pierre) [reply apps/easy-messe/components/OfferMass/DateTimeMass/DateTimeMass.tsx:78]: Okey

File: apps/easy-messe/pages/offer-mass/index.tsx [hunk 7]
```diff
@@ -87,14 +105,35 @@ export default function OrderMass() {
...
+                                key={index}
+                                badgeContent={badgeContent}
+                            >
+                                <Tab
+                                    disableRipple
+                                    label={title}
+                                />
+                            </Badge>
+                        ))
+                    } */}
```
Comment (Tchakoumi) [inline apps/easy-messe/pages/offer-mass/index.tsx]: remove commented code before requesting for a review.

commented code shouldn't go to prod

File: apps/easy-messe/components/OfferMass/DateTimeMass/TimeClock.tsx [hunk 8]
```diff
@@ -23,6 +26,7 @@ function Time({
             timeStep={15}
             onChange={(newTime) => handleTime(newTime)}
             shouldDisableTime={(newTime) => !isTimeAllowed(newTime.toDate()) as boolean}
+            ampm={activeLanguage === 'fr' ? false : true}
```
Comment (Tchakoumi) [inline apps/easy-messe/components/OfferMass/DateTimeMass/TimeClock.tsx:29]: the condition already returns true or false.

```suggestion
            ampm={activeLanguage === 'fr'}
```

File: apps/easy-messe/components/OfferMass/OfferMassCard.tsx [hunk 9]
```diff
@@ -110,15 +122,18 @@ export default function OfferMassCart({ massInfos }: OfferMassCartProps): JSX.El
...
                         <Typography
                             variant="caption"
                             textAlign="center"
                             sx={{
                                 fontWeight: 'bold',
+                                maxWidth: { laptop: '60px', mobile: 'auto' },
                             }}
                         >
-                            {mass.massInfos.price !== null && `${mass.massInfos.price} frc`}
+                            {mass.massInfos.price !== null && `${mass.massInfos.price} frs CFA`}
```
Comment (Tchakoumi) [inline apps/easy-messe/components/OfferMass/OfferMassCard.tsx:136]: you don't need to present the number with ~frs CFA~
`formatNumber` from `react-intl` does that for you and automatically formats the number with spaces or commas, and adds the appropriate currency.

```ts
formatNumber(mass.massInfos.price, {style: 'currency', currency: 'XAF'})
```

destructure the mass object at the top of the file.

to check for `null` or `undefine` or `empty string` or `0`, you don't need to precise those, unless you need that specific check.


```suggestion
                            {mass.massInfos.price && formatNumber(mass.massInfos.price, {style: 'currency', currency: 'XAF'})}
```

**Review by Tchakoumi — CHANGES_REQUESTED** [review]

File: libs/theme/src/offerMasses/offerMass.interface.ts [hunk 10]
```diff
@@ -5,8 +5,8 @@ export interface OfferMassContextProviderProps {
 }
 
 interface Requester {
-  name: string | null;
-  phone: string | null;
+  name: string;
+  phone: string;
   anonymous: boolean;
 }
```
[the remark this answers is not in the document]
Comment (Poukam-Pierre) [reply libs/theme/src/offerMasses/offerMass.interface.ts]: All right!

File: apps/easy-messe/components/OfferMass/OfferList.tsx [hunk 11]
```diff
@@ -0,0 +1,59 @@
...
+                alignItems: 'center',
+                columnGap: '10px'
+            }}>
+                <Button
+                    variant="contained"
+                    disableElevation
+                    sx={{
+                        width: 'fit-content',
+                    }}
+                    disabled={massRequested.length === 0 ? true : false}
```
Comment (Poukam-Pierre) [inline apps/easy-messe/components/OfferMass/OfferList.tsx:44]: This test get true or false. Don't need to specify like that. 
```suggestion
                    disabled={massRequested.length === 0 }
```

File: apps/easy-messe/components/OfferMass/OfferList.tsx [hunk 12]
```diff
@@ -0,0 +1,59 @@
...
+                >{formatMessage({ id: 'souscribe' })}</Button>
+                <Typography
+                    variant="h5"
+                    sx={{
+                        paddingBottom: 0,
+                        fontWeight: 'bold',
+                        display: massRequested.length === 0 ? 'none' : 'inherit'
```
Comment (Poukam-Pierre) [inline apps/easy-messe/components/OfferMass/OfferList.tsx:51]: This has been notify above. Don't need repete the result.

File: apps/easy-messe/components/layout/Header/Header.tsx [hunk 13]
```diff
@@ -69,7 +77,7 @@ export default function Header() {
                     <IconButton sx={{
                         display: { laptop: 'none', mobile: 'block' }
                     }}
-                        onClick={() => setIsSideBarOpen(true)}
+                        onClick={() => openSideBar()}
```
Comment (Poukam-Pierre) [inline apps/easy-messe/components/layout/Header/Header.tsx:80]: Don't need arrow function here just call it.

File: apps/easy-messe/components/layout/Header/SideBar/SideBar.tsx [hunk 14]
```diff
@@ -9,16 +9,18 @@ import { NavItem } from "../navitem";
...
+
+export default function SideBar({ open, onClose, navItems }: sideBarProps) {
+    const { push, asPath } = useRouter()
     const { formatMessage } = useIntl()
+    const isActive = asPath.startsWith('/offer-mass')
     return (
         <Drawer
             open={open}
-            onClose={() => closeSideBar(false)}
+            onClose={() => onClose()}
```
Comment (Poukam-Pierre) [inline apps/easy-messe/components/layout/Header/SideBar/SideBar.tsx:23]: Don't need arrow function here. Just call it.

File: libs/theme/src/offerMasses/offerMass.interface.ts [hunk 15]
```diff
@@ -0,0 +1,30 @@
+import { Dayjs } from 'dayjs';
+
+export interface OfferMassContextProviderProps {
+  children: JSX.Element;
+}
+
+interface Requester {
```
Comment (Poukam-Pierre) [inline libs/theme/src/offerMasses/offerMass.interface.ts:7]: restructure requester's data here to make it either a person nor anonymous not having both at the same type.

File: libs/utils/src/lib/utils.ts [hunk 16]
```diff
@@ -112,3 +114,11 @@ export function getFirstTwoNameInitials(name: string) {
     .map((_) => _[0].toUpperCase())
     .join('');
 }
+
+/**
+ * Format a date and time elements from a dateTime value into a specific one as "dd month year - hh:mm"
+ * @param {Dayjs} dateTime - the date and time for a mass
+ */
+export function formattedDateTime(dateTime: Dayjs): string {
```
Comment (Poukam-Pierre) [inline libs/utils/src/lib/utils.ts:122]: consider language when figure out the format date and time.

**Review by Poukam-Pierre — COMMENTED** [review]
breack all changes and push before merge.
Thanks!

**Pull request merged**